### PR TITLE
Adding test for assert_equal_index with empty iterables

### LINKED ISSUE #16844: factorize_from_iterables applies type to empty iterables
#### Code Sample, a copy-pastable example if possible

```python
In [1]: import pandas as pd

In [2]: A = pd.MultiIndex(levels=[[], []], labels=[[], []], names=['a', 'b'])

In [3]: B = pd.MultiIndex.from_arrays(arrays=[[], []], names=['a', 'b'])

In [4]: A
Out[4]:
MultiIndex(levels=[[], []],
           labels=[[], []],
           names=[u'a', u'b'])

In [5]: B
Out[5]:
MultiIndex(levels=[[], []],
           labels=[[], []],
           names=[u'a', u'b'])

In [6]: pd.testing.assert_index_equal(A, B)
...
AssertionError: MultiIndex level [0] are different

MultiIndex level [0] classes are not equivalent
[left]:  Index([], dtype='object', name=u'a')
[right]: Float64Index([], dtype='float64', name=u'a')
```
#### Problem description

Empty iterables should be factorized to object types rather than `Float64Index`. This causes MultiIndexes created from the constructor to not equal those created using `.from_arrays`, which relies on factorization (ref PR #16782).

### PR DESCRIPTION
**Pull request by VijayantSoni**
- [x] closes #16844 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/util/test_assert_index_equal.py [hunk 1]
```diff
@@ -171,3 +171,9 @@ def test_index_equal_category_mismatch(check_categorical):
             assert_index_equal(idx1, idx2, check_categorical=check_categorical)
     else:
         assert_index_equal(idx1, idx2, check_categorical=check_categorical)
+
+
+def test_index_equal_empty_iterable():
+    a = MultiIndex(levels=[[], []], codes=[[], []], names=['a', 'b'])
```
Comment (gfyoung) [inline pandas/tests/util/test_assert_index_equal.py]: Reference issue as a comment right above.

**Comment by mroeschke** [comment]
You will need to run `black pandas`

**Review by mroeschke — COMMENTED** [review]
Can you move this test to `pandas/tests/indexes/multi/test_constructor.py`

**Comment by mroeschke** [comment]
Thanks @VijayantSoni

**Comment by VijayantSoni** [comment]
Thank you guys @mroeschke @gfyoung

**Pull request merged**